Misc. small fixes and adjustments

Daniel-Cortez opened the pull request:
**This PR does the following:**
* Optimizes Orbinaut code
  Some of the calculations are moved out of the loop and the orbs are now updated only when the player is on ground and moving (there's no reason to update them all the time if their position haven't changed since the previous physics frame).
* Optimizes checkpoint searching code in `Player.gd`.
* Fixes a bug with the checkpoint time only being restored when respawning at the last checkpoint (for every other checkpoint the time was being reset to zero).
* Fixes 1-up monitors not giving an extra life.
  It seems that I accidentally removed the `Global.lives += 1` line when implementing `MusicController`, as that line was "hidden" among the other lines of code that played the 1-up jingle.
* Makes `ShieldAnimations.gd` use named constants for blend modes, instead of hardcoded IDs.
* Fixes color inconsistencies in HUD graphics.
* Other small adjustments.

Daniel-Cortez committed:
Fly.gd: Use chained `if`/`elif`s instead of nested `if`s

Daniel-Cortez committed:
Player.gd: Move calculations for rotation angle of invincibility stars out of the loop

Daniel-Cortez committed:
WaterBar.gd: Slightly simplify calculations, use a single `clampf()` instead of a combination of `min()`/`max()`

Daniel-Cortez committed:
Monitor.gd: Only multiply the Y part of the vector

The X part remains zero anyway, so there's no need to multiply the whole vector.

Daniel-Cortez committed:
ShieldAnimations.gd: Use named constants for blend modes

Daniel-Cortez committed:
ShieldAnimations.gd: Simplify oddity checks

`frame` is already an integer, so there's no reason to use `fmod()`.

Daniel-Cortez committed:
ShieldAnimations.gd: Merge two cases for the Bubble Shield

Daniel-Cortez committed:
Orbinaut.gd: Don't calculate player's direction twice

Daniel-Cortez committed:
Orbinaut.gd: Do movement calculations and update orb positions only when the player is on ground and moving

Daniel-Cortez committed:
Monitor.gd: Don't check if we're in editor on each loop iteration, just do it once

Daniel-Cortez committed:
Don't skip index `0` inside loop body, just specify the starting index using `range()`.

When `range()` is used to specify a loop range, the loop is compiled into optimal bytecode that doesn't actually allocate an array or even call `range()`, so using that function should be zero-cost in such cases.

Daniel-Cortez committed:
NumeralDisplay.gd: Multiply vectors as whole instead of multiplying its components separately

Daniel-Cortez committed:
CNZBarrel.gd, VerticalBar.gd: Use `Vector2.ZERO` for zero vectors

Daniel-Cortez committed:
StateMoverMaster.gd: Remove explicit call of `Vector2` constructor

`Vector2` is not an object, so we don't need to call the constructor explicitly in order to create a copy.

Daniel-Cortez committed:
StateMoverMaster.gd: Use `floor()` on the whole vector instead of doing so twice for each of its components

Daniel-Cortez committed:
hud_numerals_hyper.png: Make number colors to be more consistent with the colors of the blue ring sprite

Daniel-Cortez committed:
hud_text.png: Fix white and yellow being tinted (and thus being inconsistent with white and yellow in `hud_results_text.png`)

Also fixed red and blue to be more consistent with Sonic's palette.

Daniel-Cortez committed:
Fix white being darkened in a few HUD files (`#EEEEEE` => `#FFFFFF`)

Daniel-Cortez committed:
Player.gd: Move the character assignment line to the beginning of the code block, so it would be more obvious why exactly it won't work on player 2

Daniel-Cortez committed:
Player.gd: Simplify checkpoint search code

This also fixes a bug with the checkpoint time only being restored when respawning at the last checkpoint (for every other checkpoint the time was being reset to zero).

Daniel-Cortez committed:
Monitor.gd: Actually add 1 extra life when breaking 1-up monitor

It seems that I accidentally removed that line in 1a9d35595c5aa82e8a0ad80e4de122dad307776b, as it was inbetween the lines of the old code that played the 1-up jingle.

Daniel-Cortez committed:
StateMoverMaster.gd: The comment about the `delta` parameter obviously belongs to `_process()`, not `setState()`, so move it there

Also, move `setState()` before the private functions (or "underscore-prefixed" functions, to be precise, as sadly there's no concept of private functions in GDScript), while we're at it.

Daniel-Cortez committed:
BaseZone.tscn: What was this Super monitor in the Knuckles-only section doing in the "Players" group?

Daniel-Cortez committed:
Fix duplicate UID in `CardFont.png` and `largefont1.png`

Daniel-Cortez committed:
BossBoundrySetter.gd: Simplify the code for setting a variable (we have `set_deferred()` for this)

Daniel-Cortez committed:
Slightly tidy the code up by making use of pattern guards

DimensionWarped commented:
Fix the conflicts in here and I'll give this another look and try to get it merged tomorrow.

Daniel-Cortez commented:
Done.

Daniel-Cortez committed:
HUD.gd: Fix `Sprite2D.frame` being assigned an out-of-bounds value

When loading Emerald Hill Zone, Godot was printing an error message on line `$LevelCard/Banner/Act.frame = act-1` because of `frame` being assigned `-1`, since the act is set to `0`.

Daniel-Cortez committed:
ScrewPlatform.gd: Use named constants for movement directions

Daniel-Cortez committed:
ScrewPlatform.gd: Check for falling after the position is updated

Also, this check could be performed multiple times per update tick because for some reason it was placed inside of a per-player loop, even though the position doesn't change in that loop.

Daniel-Cortez committed:
ScrewPlatform.gd: Update the position and check for falling only when actually moving (`moveOffset != 0.0`)

Daniel-Cortez committed:
ScrewPlatform.gd: Move duplicated code into a separate function

Daniel-Cortez committed:
BossBoundrySetter.gd: Fix undefined variable error (a leftover from the camera refactor)

Daniel-Cortez committed:
CharacterSelect.gd: Remove redundant assignment

Daniel-Cortez committed:
CharacterSelect.gd: Remove redundant use of a documentation comment, use a normal one instead

Daniel-Cortez committed:
ConveyorBelt.gd: Simplify calculations in `_process()`, move duplicate code into a separate function

Daniel-Cortez committed:
Player.gd: Documentation comments overhaul.
* Makes better use of documentation tags.
* Adds missing commas at the ends of sentences.
* Other minor adjustments.

Didn't touch the comments for `calculate_input()`, as I'm not sure what to do with it: the function is completely unused, and GlobalFunctions.gd also has an exact copy of this function.

Daniel-Cortez committed:
Player.gd: Add a documentation comment for `get_camera()`

Daniel-Cortez committed:
ConnectableGimmick.gd: Make better use of documentation tags.

Daniel-Cortez committed:
ConnectableGimmick.gd: Add a comment about using `@abstract`

Daniel-Cortez committed:
Replace direct references to `shield` and `partner` with `get_shield()` and `get_partner()`

Daniel-Cortez committed:
Player.gd: Rename `partner` into `_partner`, so it won't appear in editor suggestions for autocompletion

Daniel-Cortez committed:
Player.gd: Rename `shield` into `_shield`

Daniel-Cortez committed:
Replace all uses of `groundSpeed` with `get_ground_speed()` and `set_ground_speed()`

Daniel-Cortez committed:
Player.gd: Rename `groundSpeed` into `_ground_speed`

Daniel-Cortez committed:
Player.gd: Implement `set_direction_signed()` and `flip_direction()`

Daniel-Cortez committed:
Replace direct use of `PlayerChar.direction` with use of accessor functions

Daniel-Cortez committed:
Player.gd: Rename `direction` into `_direction`
